nilearn/tests/test_datasets.py

```diff
@@ -27,14 +27,22 @@
 file_mock = None
 
 
+def setup_silence():
+    # Wrap functions with versions that have verbose=0
+    datasets._get_dataset_dir = silence_fn(datasets._get_dataset_dir)
+    datasets._uncompress_file = silence_fn(datasets._uncompress_file)
+    datasets._fetch_file = silence_fn(datasets._fetch_file)
+    datasets._fetch_files = silence_fn(datasets._fetch_files)
+    datasets._chunk_read_ = silence_fn(datasets._chunk_read_)
```

Comment:
tl; dr: it just seems to me that passing `verbose=0` into the needed functions is a lot simpler.

It does the job we want which is to reduce noise in the output of nosetests. Yes we may forget to set `verbose=0` occasionally when adding a new test, but we will probably notice eventually when either running the tests or looking at the travis-ci output.

Also as I mentioned above, it could be a way to potentially detect bugs, like the verbose parameters that aren't passed down the call stack. Let's be honest, I don't think we have so much time on our hands that we are going to unit test each of our print statements.

> To test the output directly, just don't add the decorator to the test

That's actually not true because you don't have a teardown method and you only need one decorated test with setup_silence to silence your dataset functions for all the tests that are going to run from then on. Changing the global state in tests is very very nasty.

An example of the side-effects, add this test at the end of test_datasets.py that checks that some printing on stdout actually occurs:

```python

import sys
from contextlib import contextmanager

@contextmanager
def redirect_stdout(new_target):
    old_target, sys.stdout = sys.stdout, new_target # replace sys.stdout
    try:
        yield new_target # run some code with the replaced stdout
    finally:
        sys.stdout = old_target # restore to the previous value

def test_stdout_miyawaki2008():
    with redirect_stdout(open('/tmp/test', 'w')):
        datasets._get_dataset_dir('test', verbose=5)

    content = open('/tmp/test').read()
    assert_true('search paths' in content)

```

This won't pass with you current code but it will pass if you remove your setup_silence decorators (or have setup_silence be just a no-op)